compiler/shared_ast/typing.ml

```diff
@@ -1107,3 +1098,7 @@ let program ?fail_on_any ?assume_op_types prg =
             prg.decl_ctx.ctx_enums;
       };
   }
+
+let program ?fail_on_any ?assume_op_types prg =
+  Message.with_delayed_errors (fun () ->
+      program ?fail_on_any ?assume_op_types prg)
```

Comment:
Can there be an option to delay the errors or not if we want to?

Reply:
After discussions, I also added an optional argument that defaults to the CLI option.